### DISCUSSION TITLE
How work the upload chain ?

### DISCUSSION THREAD

**Discussion by BobSquarePants** [pallets/flask#5143] [Q&A] [answered]
Hi everyone,

I would like to secure as possible the upload process. Therefore I need to get how work the upload chain in this scenario 

> nginx -> gunicorn -> flask

I guess nginx will be the first to handle the incoming `form enctype="multipart/form-data"`

in nginx we can use for example:

```nginx
location /uploads {
 ...
 client_max_body_size 100M;
 } 
```
Doing so will flask will be still able to handle the uploaded file ?

Thanks..

**Comment by davidism** [comment, marked as answer]
Yes

**Comment by BobSquarePants** [comment]
Thank you @davidism ,

when I'm using in nginx conf file

```nginx
location /uploads
```
I get a `404` error from werkzeug.. when I try to reach that endpoint.

I've took the same declaration that in my `/`

> proxy_pass

etc.. but still not working , is it possible use many nginx location with flask ? if yes how ? :)

Thanks.

**Reply by BobSquarePants** [reply]
I found :p

my entry in nginx

```nginx
location /uploads {
proxy_pass http://127.0.0.1:12345/
}
```
was wrong had should point to `http://127.0.0.1:12345/uploads/`

now it work like a charm.
